api/metrics/renderers.py

```diff
@@ -42,10 +45,19 @@ def get_csv_row(keys_list, report_attrs):
     ]
 
 
-class MetricsReportsCsvRenderer(renderers.BaseRenderer):
-    media_type = 'text/csv'
-    format = 'csv'
-    CSV_DIALECT = csv.excel
+class MetricsReportsRenderer(renderers.BaseRenderer):
+
+    def get_filename(self, renderer_context: dict, format_type: str) -> str:
+        """Generate the filename for the file based on format_type REPORT_FILENAME_FORMAT and current date."""
+        if renderer_context and 'view' in renderer_context:
+            current_date = datetime.datetime.now().strftime('%Y-%m')
+            return REPORT_FILENAME_FORMAT.format(
+                view_name=renderer_context['view'].view_name,
+                date_created=current_date,
+                format_type=format_type,
+            )
+        else:
+            raise NotImplementedError('Missing format filename')
 
     def render(self, json_response, accepted_media_type=None, renderer_context=None):
```

Comment:
code smell: the base (non-csv-specific) `MetricsReportsRenderer` now has this `render` implementation that only makes sense on csv-specific base classes -- recommend moving the csv-`render` to either of the csv/tsv renderers and sharing logic that way